transport/server.cc

```diff
@@ -236,23 +238,49 @@ cql_server::do_accepts(int which, bool keepalive, socket_address server_addr) {
             auto conn = make_shared<connection>(*this, server_addr, std::move(fd), std::move(addr));
             ++_connects;
             ++_connections;
-            // Move connection to the background, monitor for lifetime and errors.
-            (void)conn->process().then_wrapped([this, conn] (future<> f) {
-                --_connections;
-                try {
-                    f.get();
-                } catch (...) {
-                    clogger.debug("connection error: {}", std::current_exception());
-                }
+            return advertise_new_connection(conn).then([this, conn] {
+                // Move connection to the background, monitor for lifetime and errors.
+                (void)conn->process().then_wrapped([this, conn] (future<> f) {
+                    try {
```

Comment:
I think the (void) should be moved to the caller. The accept loop shouldn't wait for advertise_new_connection(), and shouldn't fail if a_n_c() fails.